decidim-core/app/helpers/decidim/widget_urls_helper.rb

```diff
@@ -2,8 +2,8 @@
 module Decidim
   module WidgetUrlsHelper
     def embed_modal_for(url)
-      js_embed_code = "#{content_tag(:script, '', src: url)}"
-      embed_code = "#{content_tag(:noscript, content_tag(:iframe, '', src: url.gsub(".js", ".html"), frameborder: 0, scrolling: "vertical"))}"
+      js_embed_code = content_tag(:script, "", src: url).to_s
```

Comment:
We tried `String.new` too, but we went for `to_str` because it was shorter 😆 But it might lead to confusion, someone might think it's a typo and change it to `to_s` which will break things, so `String.new` wins here (apart from being more resilient as you pointed out). So go for it and please add a comment explaining why we need to turn this into a String (so that it doesn't render the HTML).

Another option would be writing the raw HTML instead of using `content_for`, but seems more error-prone to me 😕